llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -1255,6 +1258,12 @@ void RISCVInsertVSETVLI::emitVSETVLIs(MachineBasicBlock &MBB) {
         MachineOperand &VLOp = MI.getOperand(getVLOpNum(MI));
         if (VLOp.isReg()) {
           // Erase the AVL operand from the instruction.
+          if (MachineInstr *MI = MRI->getVRegDef(VLOp.getReg());
+              MI && MI->getOpcode() == RISCV::ADDI &&
```

Comment:
Technically ADDI can have a non-registeroperand 1 or a non-immediate operand 2. It's very unlikely to happen since that should only happen with pointers, but someone could do a strange bitcast. We should check isReg() and isImm() before getReg and getImm to be safe.

Reply:
We should probably extract the code from `hasNonZeroAVL` so we can share it.